components/Fields/RoutesField.vue

```diff
@@ -43,79 +37,31 @@ export default defineNuxtComponent({
     },
   },
 
-  computed: {
-    propertyTranslations() {
-      return this.$propertyTranslations
-    },
-
-    isCompact(): boolean {
-      return this.context === PropertyTranslationsContextEnum.Card
-    },
-
-    routes(): { [key: string]: Route } {
-      const activitiesStruct: { [key: string]: { [key: string]: string } } = {}
-      Object.entries(this.properties || {})
-        .map(([key, value]) => [key.split(':'), value])
-        .filter(([keys, _value]) => keys[0] === 'route' && keys.length === 3)
-        .forEach(([[_, activity, property], value]) => {
-          if (activitiesStruct[activity]) {
-            activitiesStruct[activity][property] = value
-          }
-          else {
-            activitiesStruct[activity] = {}
-            activitiesStruct[activity][property] = value
-          }
-        })
-
-      const ret: { [key: string]: Route } = {}
-      Object.entries(activitiesStruct).forEach(([acivity, properties]) => {
-        ret[acivity] = properties
-      })
-      return ret
-    },
-
-    length(): string | undefined {
-      const route = Object.values(this.routes)[0]
-      return route?.length
-        ? this.$t('units.km', { length: route.length })
-        : undefined
-    },
+  setup() {
+    const {
+      getRouteDifficulty,
+      getRouteDuration,
+      getRouteLength,
+      getRouteNoDetails,
+      getRoutes,
+    } = useField()
+
+    return {
+      getRouteDifficulty,
+      getRouteDuration,
+      getRouteLength,
+      getRouteNoDetails,
+      getRoutes,
+    }
   },
 
-  methods: {
-    duration(route: Route): string | undefined {
-      if (route.duration) {
-        const hours = Math.floor(route.duration / 60)
-        const minutes = route.duration % 60
-
-        let string = ''
-        if (hours > 0)
-          string += this.$t('units.hours', { hours })
-
-        if (minutes > 0)
-          string += `${hours > 0 ? ' ' : ''}${this.$t('units.min', { minutes })}`
-
-        return string
-      }
-      else {
-        return undefined
-      }
-    },
-
-    difficulty(activity: string, route: Route): string | undefined {
-      return route.difficulty
-        ? this.$propertyTranslations.pv(
-            `route:${activity}:difficulty`,
-            route.difficulty,
-            this.context,
-        )
-        : undefined
+  computed: {
+    isCompact() {
```

Comment:
Could be a prop instead of computed, the reactivity would've be the same and we save an import